tests/func/test_data_cloud.py

```diff
@@ -41,6 +41,14 @@
     )
 ]
 
+# Clouds that implement the general methods that can be tested
+# for functional tests that require extensive apis (e.g traversing
+# via walk_files())
+full_clouds = [
+    pytest.lazy_fixture(cloud)
+    for cloud in ["s3", "gs", "azure", "ssh", "hdfs"]
+]
+
```

Comment:
Let's add these directly to `test_pull_00_prefix` parametrization, since they are not used anywhere else.